### PR TITLE
[improve][broker] re-elect the channel owner if no channel owner is found

### PR DESCRIPTION
**Pull request by heesung-sn**
…

### Motivation
- There might be a chance that there is no channel owner. The existence of the channel owner(leader) needs to be monitored.

### Modifications

- Elect the channel owner(leader) if no channel owner is found in the role(leader|follower) play and monitor logic.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java [hunk 1]
```diff
@@ -826,6 +826,22 @@ public static boolean isInternalTopic(String topic) {
                 || topic.startsWith(TOP_BUNDLES_LOAD_DATA_STORE_TOPIC);
     }
 
+    private boolean handleNoChannelOwnerError(Throwable e) {
```
Comment (Demogorgon314) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:829]: We can add the retry logic to `LeaderElectionService` later.

**Review by Demogorgon314 — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java (hunk 1 above)
Comment (heesung-sn) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:829]: yes.

**Comment by lhotari** [comment]
@heesung-sn is there a need to backport this change?

**Comment by heesung-sn** [comment]
Do we still maintain 3.x branches? I thought we maintain 4.x only now.

We can backport this to 3.3, too.

**Comment by lhotari** [comment]
> Do we still maintain 3.x branches? I thought we maintain 4.x only now.
> 
> We can backport this to 3.3 and 4.0.

We follow the schedule available at https://pulsar.apache.org/contribute/release-policy/#supported-versions . branch-3.3 will stop being in active maintenance in about 1 month. 3.0.x is maintained until May 2025.

**Comment by heesung-sn** [comment]
cherry-pick pr to 3.3 https://github.com/apache/pulsar/pull/23580

**Pull request merged**